packages/x-charts/src/hooks/animation/useAnimate.ts

```diff
@@ -0,0 +1,65 @@
+import useForkRef from '@mui/utils/useForkRef';
+import * as React from 'react';
+import { useAnimateInternal } from '../../internals/animation/useAnimateInternal';
+
+export interface UseAnimateParams<Props extends {}, Elem extends Element, T extends {} = Props> {
+  createInterpolator: (lastProps: Props, newProps: Props) => (t: number) => Props;
+  transformProps: (props: Props) => T;
+  applyProps: (element: Elem, props: T) => void;
+  skip?: boolean;
+  initialProps?: Props;
+  ref?: React.Ref<Elem>;
+}
+
+export type UseAnimateReturn<Elem extends Element, T extends {}> = Omit<T, 'ref'> & {
+  ref: React.Ref<Elem>;
+};
+
+/** Animates a ref from {@link initialProps} to {@link props}. If {@link props} change, the animation will resume towards
+ * the new {@link props}. Use this hook if you want to customize the animation of an element.
+ *
```

Comment:
```suggestion
/**
 * Animates a ref from {@link initialProps} to {@link props}.
 * If {@link props} change, the animation will resume towards the new {@link props}.
 * Use this hook if you want to customize the animation of an element.
 *
```

Overall it looks like documenting this hook in one block is a bit hard.
Might be simpler to document it parameter per parameter